### PR TITLE
subscribe() returns unsubscriber

### PR DESCRIPTION
**Pull request by ProLoser**
After poking around the socket.io source code, found removeListener (wonderfully undocumented)

### CODE REVIEW

File: src/socketBackend.js [hunk 1]
```diff
@@ -48,7 +48,9 @@ function createSailsBackend($browser, $window, $injector, $q, $timeout){
 
     //TODO normalize http paths to event names
     connection.subscribe = function(event,handler){
-        $window.io.socket.on(event,tick($window.io.socket,handler));
+        var callback = tick($window.io.socket,handler);
+        $window.io.socket.on(event,callback);
```
Comment (thebigredgeek) [inline src/socketBackend.js]: Why did you change the signature here, @ProLoser ?

**Comment by thebigredgeek** [comment]
Functionally, the docs also need to be updated to represent the change made.  Otherwise no one will know that it returns an `unbind` function.

File: src/socketBackend.js (hunk 1 above)
Comment (ProLoser) [inline src/socketBackend.js]: So that you can unsubscribe.

**Comment by sgress454** [comment]
This looks good to me, thanks @ProLoser!  Added docs, but I'm not sure how the http://balderdashy.github.io/angularSails site gets regenerated; @robwormald?
